### PR TITLE
Longer list and deep linking

### PR DESCRIPTION
**Pull request by marcellourbani**
Tomasz,
I have three changes here:
1) deep linking
2) smaller font in issue list, show last 4 char
3) dummy changes to get it to build in my version of android studio.
I think the first might interest you, it's a bit unpolished (see below) but works, the rest is your call.
Notes on the deep link: 
-first version removed the account list from history, but then it was VERY tricky to change account or even go to an issue list. 
-couldn't find a way to make it react to specific URLs it didn't know in advance. So it will be proposed for any URL who matches https:*browse/*
-if you enter with an url for which you don't have an account it tells you in a dialog, but doesn't show the add account activity with the missing prefix

### CODE REVIEW

**Comment by ikogan** [comment]
I love the deep linking idea, but the broadness of the intent filter concerns me. Is there a way we can make it something like https?://.*/browse/[a-zA-Z0-9]+-[0-9]+ or something similar? Effectively whatever Jira's regex is for issue IDs.

**Comment by marcellourbani** [comment]
Didn't try that for 2 reasons :
1) not sure what is the actual jira regex
2) looks like it is a simple glob,  only does *,  no square brackets

Marcello

**Comment by marcellourbani** [comment]
I hate having it broad,  but can't find better other than compiling with my
prefix in full.
Don't think is a disaster though : the user is more likely to pick chrome
than us for random links.

**Comment by ikogan** [comment]
You're correct, it looks like globbing is all you've got. What if you did something like /browser/*-*?

**Comment by marcellourbani** [comment]
/browse/.*-.* works, updated in the repository.
Not sure whether I need a new pull request

File: app/src/main/java/com/tadamski/arij/issue/list/IssueListAdapter.java [hunk 1]
```diff
@@ -67,7 +67,10 @@ public boolean hasStableIds() {
 
     private void setViewProperties(View convertView, final Issue issueSummary) {
         setTextViewText(convertView, R.id.project_shortcut, issueSummary.getKey().split("-")[0]);
-        setTextViewText(convertView, R.id.issue_id_number, issueSummary.getKey().split("-")[1]);
+        String inum = issueSummary.getKey().split("-")[1];
+        //up to 4 digits. If bigger, show the LAST 4 perhaps indicating that it was clipped would be a good idea...
+        if(inum.length()>4)inum = inum.substring(inum.length()-4);
+        setTextViewText(convertView, R.id.issue_id_number, inum);
```
Comment (tmszdmsk) [inline app/src/main/java/com/tadamski/arij/issue/list/IssueListAdapter.java]: I think it can be done in the layout file.

File: app/app.iml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (tmszdmsk) [inline app/app.iml]: this file should be added to .gitignore

File: arij.iml [hunk 3]
```diff
@@ -1,6 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (tmszdmsk) [inline arij.iml]: this file should be added to .gitignore

**Comment by tmszdmsk** [comment]
Great improvement. I added some general comments. Will have a closer look in the evening (Warsaw time ;). Thanks.

**Comment by tmszdmsk** [comment]
Ok, I have few comments:
* I think that it's a good practice to put different features in separate pull requests requests (In this case there should be three of them i think)
* As i wrote in a comment, "truncation" should be defined in layout xml file
* I researched deeplinking mechanism some time ago, and yes, from what I've found it's not possible to specify patterns dynamically. 
* I'm not sure that handling of those links should be done in AccountsList activity. I would put it in separate component that listens for those intents. I'm not "Android expert" but I guess it can be done in form of intentservice.
* about the .iml files: I know i've put them there. They can be safely deleted.

Please, write if you want to work more on those features.

**Comment by marcellourbani** [comment]
Point by point:

   - agreed, not an expert on GIT and it's my first pull request ever, not
   really sure what I'm doing. Will be a bit clumsy for me as without the
   first commit (which I didn't really care about) I can't compile it.And as
   said before, I'm not very used to GIT
   - will look into that. I remember failing a year or so ago, but will try
   again :)
   - yep
   - Less expert than you, I do SAP for a living. I know View intents can't
   be catched by broadcastreceivers, can try with an intentservice, but the
   main reason why I used AccountList is UI related: if the first screen we
   land in is the issue, there's no obvious way to go back to the account
   list, if you close the app and open it again you land in the issue unless
   you kill it first.Perhaps we can use a service and then hijack the back
   button in IssueActivity.homeSelected(), but I don't like it
   - sure

Last but not least: I do plan to work on this (at least to deliver the
deeplink and truncation) but I'm doing it in my spare time, don't hold your
breath :)

**Pull request closed without merging**